zipkin-cassandra/src/main/scala/com/twitter/cassie/connection/ClusterClientProvider.scala

```diff
@@ -141,7 +141,11 @@ private[cassie] class ClusterClientProvider(
         // call login
         val loginMap: java.util.Map[String, String] = HashMap("username" -> username, "password" -> password)
         val authRequest = new AuthenticationRequest(loginMap)
-        client.login(authRequest)
+        val future = client.login(authRequest)
+        future onFailure { exception =>
```

Comment:
Why are you using side-effects here? Future is already rich enough to not require you to throw exceptions. What are you trying to accomplish with throwing an exception here?